lib/file-finder.js

```diff
@@ -76,18 +95,18 @@ FileFinder.prototype.findInDirectoryOrParents = function(directory) {
 
     dirs = [];
     searched = 0;
-    name = this.fileNames[0];
-    names = Array.isArray(name) ? name : [name];
+    names = this.fileNames;
 
     (function() {
         while (directory !== child) {
             dirs[searched++] = directory;
-
-            for (var k = 0, found = false; k < names.length && !found; k++) {
-
-                if (getDirectoryEntries(directory).indexOf(names[k]) !== -1 && fs.statSync(path.resolve(directory, names[k])).isFile()) {
-                    filePath = path.resolve(directory, names[k]);
-                    return;
+            var filesMap = normalizeDirectoryEntries(getDirectoryEntries(directory), directory, names);
+            if (Object.keys(filesMap).length) {
+                for (var k = 0, found = false; k < names.length && !found; k++) {
```

Comment:
Ahh. Nice catch! `found` isn't used anymore, but I forgot to remove it. Will fix it in the evening.